homeassistant/components/vivotek/camera.py

```diff
@@ -0,0 +1,136 @@
+"""Support for Vivotek IP Cameras."""
+
+import logging
+
+import voluptuous as vol
+from libpyvivotek import VivotekCamera
+
+from homeassistant.const import (
+    CONF_NAME,
+    CONF_USERNAME,
+    CONF_PASSWORD,
+    CONF_AUTHENTICATION,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+    CONF_VERIFY_SSL,
+    CONF_IP_ADDRESS,
+)
+from homeassistant.components.camera import PLATFORM_SCHEMA, SUPPORT_STREAM, Camera
+from homeassistant.helpers import config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_STREAM_SOURCE = "stream_source"
+CONF_FRAMERATE = "framerate"
+
+DEFAULT_NAME = "Vivotek Camera"
+DEFAULT_EVENT_0_KEY = "event_i0_enable"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+        vol.Optional(CONF_STREAM_SOURCE, default=None): vol.Any(None, cv.string),
```

Comment:
Don't use default `None`. Remove the default and use `dict.get` when accessing the config option.

Reply:
What kind of string should the stream source option return? Does the user need to configure this?

Reply:
Ah yeah, good point. That can go.